apps/shared/WebView.qml

```diff
@@ -211,6 +211,14 @@ WebContainer {
                 if (!modelUrl || modelUrl != url) {
                     tabModel.updateThumbnailPath(tabId, "")
                 }
+
+                // Update the favicon for history items.
+                if (modelUrl && modelUrl != url) {
+console.log("XXXXXXXXX url changed to: " + url + ", grabbing favicon..")
+                    FaviconManager.grabIcon("history", webPage,
```

Comment:
will this be persisted across reboots etc?  hopefully so?

Reply:
Yes, it should be. The favicons are saved to file each time a new one is added. There's scope for a lot of optimisation here, which will be even more relevant now this is being used for history and not just logins and favicons (since the latter are changed quite rarely, whereas history changes frequently).

Reply:
Do you think there's value in adding a check inside grabIcon() which immediately returns if the type="history" and a favicon for that site has already been loaded, to avoid unnecessary favicon fetch+store?

Reply:
I was pondering this too. The downside is that if, at some point in the future, a site without a favicon gets one, the icon won't be updated; or if there's a snapshot instead of a favicon, the icon won't update if the site changes it's design. But, personally, I think the positives of doing this outweigh the negatives.

I wouldn't restrict the behaviour to history items either. I don't see why it shouldn't apply in all cases (currently just this and the login icons).

Reply:
I've now added this via a `if (!get(type, webPage->url().toString()).isEmpty()) return;` check in grabIcon()